internal/rpc/starknet/errors.go

```diff
@@ -5,62 +5,53 @@ type StarkNetError struct {
 	Message string
 }
 
-func (s *StarkNetError) Error() string {
+func (s StarkNetError) Error() string {
 	return s.Message
 }
 
-func NewContractNotFound() *StarkNetError {
-	return &StarkNetError{
+var (
+	UnexpectedError = StarkNetError{
+		Code:    -1,
+		Message: "Unexpected error",
+	}
+	NotImplementedError = StarkNetError{
+		Code:    -2,
+		Message: "Not implemented",
+	}
+	InvalidStorageKey = StarkNetError{
+		Code:    -3,
+		Message: "Invalid storage key",
+	}
```

Comment:
Done -> https://github.com/NethermindEth/juno/pull/360/commits/c8c650477f308449b324605ea510ccedae344a01

Reply:
What about the `-1 Unexpected error`? I think that should also fall under `-32603 Internal error`.

Reply:
`Internal error` is returned when a panic happens

Reply:
Yes, but not exclusively. 

That's just a byproduct of Go serving each request within its own Go routine which allows the application to detect when a panic has occurred and thus send an internal error to the user. If it were any other language a panic would occur and nothing would be sent back right?

So one would assume an internal error is meant for exactly this purpose. In fact here's what the HTTP specification for a `500 Internal Server Error` says (because the JSON-RPC 2.0 specification doesn't go into much detail with regards to that):

> A generic error message, given when an **unexpected** condition was encountered and no more specific message is suitable.

_(Emphasis mine)._

Wouldn't the `-32603 Internal error` be the equivalent here?